src/Microsoft.VisualStudio.Editor.Razor/DefaultEditorSettingsManager.cs

```diff
@@ -12,26 +13,95 @@ namespace Microsoft.VisualStudio.Editor.Razor
     [Export(typeof(EditorSettingsManager))]
     internal class DefaultEditorSettingsManager : EditorSettingsManager
     {
-        private readonly EditorSettingsManagerInternal _editorSettingsManager;
+        private readonly List<WeakReference<EventHandler<EditorSettingsChangedEventArgs>>> _changedHandlers;
+        private readonly object SettingsAccessorLock = new object();
+        private readonly ForegroundDispatcher _foregroundDispatcher;
+        private EditorSettings _settings;
 
         [ImportingConstructor]
-        public DefaultEditorSettingsManager(VisualStudioWorkspaceAccessor workspaceAccessor)
+        public DefaultEditorSettingsManager(ForegroundDispatcher foregroundDispatcher)
         {
-            var razorLanguageServices = workspaceAccessor.Workspace.Services.GetLanguageServices(RazorLanguage.Name);
-            _editorSettingsManager = razorLanguageServices.GetRequiredService<EditorSettingsManagerInternal>();
+            _foregroundDispatcher = foregroundDispatcher;
+            _settings = EditorSettings.Default;
+            _changedHandlers = new List<WeakReference<EventHandler<EditorSettingsChangedEventArgs>>>();
         }
 
         public override event EventHandler<EditorSettingsChangedEventArgs> Changed
         {
-            add => _editorSettingsManager.Changed += value;
-            remove => _editorSettingsManager.Changed -= value;
+            add
+            {
+                _foregroundDispatcher.AssertForegroundThread();
+
+                _changedHandlers.Add(new WeakReference<EventHandler<EditorSettingsChangedEventArgs>>(value));
+            }
+            remove
+            {
+                _foregroundDispatcher.AssertForegroundThread();
+
+                for (var i = 0; i < _changedHandlers.Count; i++)
+                {
+                    if (_changedHandlers[i].TryGetTarget(out var target) && target == value)
+                    {
+                        _changedHandlers.RemoveAt(i);
```

Comment:
This has a bug. You need to use #TheTaylor

Reply:
Nope, we return immediately no #TheTaylor necessary

Reply:
Ah I see, you're only removing one at a time.